packages/core/src/shared/ui/wizardPrompter.ts

```diff
@@ -3,19 +3,29 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { StepEstimator, Wizard } from '../wizards/wizard'
+import _ from 'lodash'
+import { StepEstimator, Wizard, WIZARD_BACK, WIZARD_SKIP } from '../wizards/wizard'
 import { Prompter, PromptResult } from './prompter'
 
 /**
  * Wraps {@link Wizard} object into its own {@link Prompter}, allowing wizards to use other
  * wizards in their flows.
+ *
+ * @remarks
+ *  - This class should only be used inside wizard classes that extend {@link NestedWizard}.
```

Comment:
Oh, this may be a good reason to keep the NestedWizard class: it makes this constraint easier to explain. Though I wonder why this constraint was added?

Reply:
Because nested class have the memorizing logic for restoring child wizard state.

Reply:
Maybe the wording of the comment is unclear to me. Does "This class" refer to WizardPrompter? For example, currently WizardPrompter is used here (which is not "inside NestedWizard"): https://github.com/aws/aws-toolkit-vscode/blob/e63079d3d3af18a3af27914e842d915f307b38b3/packages/core/src/awsService/apprunner/wizards/imageRepositoryWizard.ts#L237

Reply:
Updated the doc string.